File: MatchDetails.py
import requests

apiKey = 'RGAPI-da221bc0-65ee-4894-9b93-3b7653c963db'
region = 'na1'
summonerName = 'AYD Instinct'

url = f'https://{region}.api.riotgames.com/lol/summoner/v4/summoners/by-name/{summonerName}?api_key={apiKey}'
response = requests.get(url)
data = response.json()
puuid = data['puuid']

url2 = f'https://americas.api.riotgames.com/lol/match/v5/matches/by-puuid/{puuid}/ids?start=0&count=20&api_key={apiKey}'
response2 = requests.get(url2)
matchIds = response2.json()

counter = 0

for matchId in matchIds:
    if counter < 1:
        url3 = f'https://americas.api.riotgames.com/lol/match/v5/matches/{matchId}?api_key={apiKey}'
        response3 = requests.get(url3)
        data3 = response3.json()
        participants = data3['metadata']['participants']
        teams = data3['info']['teams']
        print(data3)
        print(teams)
    counter += 1

File: match_details.js
const apiKey = 'RGAPI-e1ce4bc1-1d8e-4bc6-8add-168fd0b78990';
const region = 'na1';
//const summonerName = '7u87p1Czhiuq';
const summonerName = 'AYD Instinct';

const url = `https://${region}.api.riotgames.com/lol/summoner/v4/summoners/by-name/${summonerName}?api_key=${apiKey}`;
//console.log(url);
fetch(url)
  .then(response => response.json())
  .then(data => {
    const puuid = data.puuid;
    //console.log(puuid);

    const url2 = `https://americas.api.riotgames.com/lol/match/v5/matches/by-puuid/${puuid}/ids?start=0&count=20&api_key=${apiKey}`;
    return fetch(url2);
  })
  .then(response => response.json())
  .then(data => {
    const matchIds = data;
    //console.log(matchIds);

    let counter = 0;
    matchIds.forEach(matchId => {
      if (counter < 1) {
        const url3 = `https://americas.api.riotgames.com/lol/match/v5/matches/${matchId}?api_key=${apiKey}`;
        fetch(url3)
        .then(response => response.json())   
        .then(data => {
          const participants = data.metadata.participants;
          const teams = data.info.teams;
          console.log(data);
          console.log(teams);

        })
      }
      counter++;
    })
  })
  .catch(error => console.error(error))


